tests/serve_request.rs

```diff
@@ -260,3 +260,18 @@ fn serve_index_instead_of_404_in_spa_mode(
 
     Ok(())
 }
+
+#[rstest]
+fn serves_requests_with_path_prefix(
+    #[with(["--route-prefix", "foobar"])] server: TestServer,
+) -> Result<(), Error> {
```

Comment:
On the old pr @aliemjay mentioned having more test cases. I don't really understand that case macro, the first two booleans make me wonder. Can someone help me with this?